### PR TITLE
[TIMOB-26196] iOS: Remove SCListener, use AVAudioRecorder instead

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26196

> `no tests` since Jenkins can probably not speak to the Simulator. We could add type-tests for the properties, but that didn't really change.

### CODE REVIEW

File: iphone/Classes/MediaModule.m [hunk 1]
```diff
@@ -670,28 +669,71 @@ - (void)requestAudioRecorderPermissions:(id)args
 
 - (void)startMicrophoneMonitor:(id)args
 {
-  [[SCListener sharedListener] listen];
+  if (_microphoneRecorder != nil) {
+    DebugLog(@"[ERROR] Trying to start new microphone monitoring while the old one is still active. Call \"stopMicrophoneMonitor()\" before and try again.");
+    return;
+  }
+
+  NSURL *url = [NSURL fileURLWithPath:@"/dev/null"];
```
Comment (vijaysingh-axway) [inline iphone/Classes/MediaModule.m]: Is value @"/dev/null" is correct for path or we should get path using [NSFileMange defaultManage] api to get path?

File: iphone/Classes/MediaModule.m [hunk 2]
```diff
@@ -670,28 +669,71 @@ - (void)requestAudioRecorderPermissions:(id)args
...
+  NSDictionary *settings = [NSDictionary dictionaryWithObjectsAndKeys:
+                                             [NSNumber numberWithFloat:44100.0], AVSampleRateKey,
+                                         [NSNumber numberWithInt:kAudioFormatAppleLossless], AVFormatIDKey,
+                                         [NSNumber numberWithInt:1], AVNumberOfChannelsKey,
+                                         [NSNumber numberWithInt:AVAudioQualityMax], AVEncoderAudioQualityKey,
+                                         nil];
+
```
Comment (vijaysingh-axway) [inline iphone/Classes/MediaModule.m]: Are these default value of setting ? Is there any doc for recommending these settings?

File: iphone/Classes/MediaModule.m [hunk 3]
```diff
@@ -670,28 +669,71 @@ - (void)requestAudioRecorderPermissions:(id)args
...
+  NSError *error;
+
+  _microphoneRecorder = [[AVAudioRecorder alloc] initWithURL:url settings:settings error:&error];
+
```
Comment (vijaysingh-axway) [inline iphone/Classes/MediaModule.m]: Should check for permission 'NSMicrophoneUsageDescription'

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
Can you please address the comments. Thanks

File: iphone/Classes/MediaModule.m (hunk 2 above)
Comment (hansemannn) [reply iphone/Classes/MediaModule.m]: They are recommended for backwards compatibility with the old API. Especially the sample rate (44100.0) is for that. The other ones ensure a certain quality of the recording.

File: iphone/Classes/MediaModule.m (hunk 3 above)
Comment (hansemannn) [reply iphone/Classes/MediaModule.m]: We can do that, yes. But that was not part of the old API as well, so it would only be a log-warning.

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Pull request merged**